simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/component/StationComponent.java

```diff
@@ -94,4 +96,23 @@ public SetVariableValidationResult validate(SetVariableDatum setVariableDatum) {
         return new SetVariableValidationResult(setVariableDatum, validationResult);
     }
 
+    /**
+     * Generates report data for all variable in the component
+     *
+     * @param onlyMutableVariables if true, returns only those variables that can be set by the operator
+     * @return list of {@link ReportDatum}
+     */
+    public List<ReportDatum> generateReportData(boolean onlyMutableVariables) {
+        List<ReportDatum> reportData = new ArrayList<>();
+        variableAccessors.values().forEach(accessor -> {
```

Comment:
I think it would look more clear if you'd use .filter().map() expression

Reply:
ok i guess it's a personal preference then

`components.values().stream().flatMap(component -> component.generateReportData(onlyMutableVariables).stream()).collect(toList());`

vs

`List<ReportDatum> reportData = new ArrayList<>();
        components.values().forEach(component -> reportData.addAll(component.generateReportData(onlyMutableVariables)));`

but i'll change it to streams (even though it's longer than a simple for loop)